src/praisonai-agents/praisonaiagents/agents/autoagents.py

```
@@ -237,84 +265,164 @@ def _generate_config(self) -> AutoAgentsConfig:
-                    if cleaned_response.endswith("```"):
-                        cleaned_response = cleaned_response[:-3]
-                    cleaned_response = cleaned_response.strip()
+                    # Check if we have OpenAI API and the model supports structured output
+                    if self.llm and (self.llm.startswith('gpt-') or self.llm.startswith('o1-') or self.llm.startswith('o3-')):
+                        # Create a new client instance if custom parameters are provided
+                        if self.api_key or self.base_url:
+                            client = OpenAIClient(api_key=self.api_key, base_url=self.base_url)
+                        else:
+                            client = get_openai_client()
+                        use_openai_structured = True
+                except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)